exporter/influxdbexporter/writer.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package influxdbexporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb-observability/common"
+	lineprotocol "github.com/influxdata/line-protocol/v2/influxdata"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+type influxHTTPWriter struct {
+	encoderPool sync.Pool
+	httpClient  *http.Client
+	writeURL    string
+
+	logger common.Logger
+}
+
+func newInfluxHTTPWriter(logger common.Logger, config *Config) (*influxHTTPWriter, error) {
+	writeURL, err := url.Parse(config.HTTPClientSettings.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+	if writeURL.Path == "" || writeURL.Path == "/" {
+		writeURL, err = writeURL.Parse("api/v2/write")
+		if err != nil {
+			return nil, err
+		}
+	}
+	queryValues := writeURL.Query()
+	queryValues.Set("org", config.Org)
+	queryValues.Set("bucket", config.Bucket)
+	queryValues.Set("precision", "ns")
+	writeURL.RawQuery = queryValues.Encode()
+
+	if config.Token != "" {
+		config.HTTPClientSettings.Headers["Authorization"] = "Token " + config.Token
```

Comment:
It may be worth saying in the README that "token" will take precedence over "headers.Authorization" if both are specified.